tests/sources/fixtures/fixture.py

```diff
@@ -76,62 +84,118 @@ def _parser():
+                    return await func(*args, **kwargs)
+                except ConnectionTimeout:
+                    if retry >= retries:
+                        raise
+                    retry += 1
+                except ApiError as e:
+                    if e.status_code != 429:
+                        raise
+                    if retry >= retries:
+                        raise
+                    retry += 1
+                finally:
+                    # TODO: 15 is arbitrary
+                    # Default retry time is 15 * SUM(1, 5) = 225 seconds, should be enough as a start
+                    time_to_sleep = time_to_sleep_between_retries(
+                        RetryStrategy.LINEAR_BACKOFF, 15, retry
+                    )
+                    await asyncio.sleep(time_to_sleep)
+
+        return wrapped
+
+    return wrapper
+
+
 def _es_client():
     options = {
-        "hosts": ["http://127.0.0.1:9200"],
-        "basic_auth": ("elastic", "changeme"),
+        "host": "http://127.0.0.1:9200",
+        "username": "elastic",
+        "password": "changeme",
     }
-    return Elasticsearch(**options)
+    return ESClient(options)
 
 
-def _monitor_service(pid):
-    es_client = _es_client()
-    sync_job_timeout = 20 * 60  # 20 minutes timeout
+@retrying_transient_errors()
+async def _wait_for_connectors_index(es_client):
+    start = time.time()
     index_present_timeout = 30  # 30 seconds
+    waiting_for_connectors_index = True
+    while waiting_for_connectors_index:
+        if await es_client.index_exists(CONNECTORS_INDEX):
+            logger.info(f"{CONNECTORS_INDEX} detected")
+            waiting_for_connectors_index = False
+            break
+        else:
+            if time.time() - start > index_present_timeout:
+                msg = f"{CONNECTORS_INDEX} not present after {index_present_timeout} seconds."
+                raise Exception(msg)
 
-    try:
-        # we should have something like connectorIndex.search()[0].last_synced
-        # once we have ConnectorIndex and Connector class ready
-        start = time.time()
-        while True:
-            response = es_client.search(index=CONNECTORS_INDEX, size=1)
+            logger.info(f"{CONNECTORS_INDEX} not present, waiting...")
+            await asyncio.sleep(1)
 
-            if len(response["hits"]["hits"]) == 0:
-                if time.time() - start > index_present_timeout:
-                    msg = f"{CONNECTORS_INDEX} not present after {index_present_timeout} seconds."
-                    raise Exception(msg)
 
-                logger.info(f"{CONNECTORS_INDEX} not present, waiting...")
-                time.sleep(1)
-            else:
-                logger.info(f"{CONNECTORS_INDEX} detected")
-                break
+@retrying_transient_errors()
+async def _fetch_connector_metadata(es_client):
+    # There's only one connector, so we just fetch it always
+    response = await es_client.client.search(index=CONNECTORS_INDEX, size=1)
```

Comment:
this may come back to bite us one day. What if we did `size=2` and raise an exception if `response["hits"]["hits"].size > 1`? That way we can track down where this assumption lives if/when we break it.

Reply:
Yeah I can do this, I tried just keeping original code without modifications in this PR (thus, not fixing such things), but I think this one is small enough to address